src/helpers/byte-array.ts

```diff
@@ -21,9 +21,12 @@ import { toHex, fromHex } from '@cosmjs/encoding';
   return 0;
 }
 
-
-export function toNumber(byteArray: Uint8Array) {
-  return parseInt(toHex(byteArray));
+export function toNumber(byteArray: Uint8Array | number[]) {
```

Comment:
a comment telling us it's big endian would be nice
alternatively a flag argument to switch between big & lil endian